From 5536d2e2cab6a9830333e18cab4b69d601358c66 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Wed, 13 Jun 2018 19:00:21 +0000
Subject: [PATCH] TRITON-500 Use batch() in writeObjects() for VMAPI tests
 Reviewed by: Josh Wilsdon <jwilsdon@joyent.com> Reviewed by: Trent Mick
 <trentm@gmail.com> Approved by: Trent Mick <trentm@gmail.com>

---
 test/lib/moray.js | 40 ++++++++++++++++++++++------------------
 1 file changed, 22 insertions(+), 18 deletions(-)

diff --git a/test/lib/moray.js b/test/lib/moray.js
index b0809a8..6d8f71e 100644
--- a/test/lib/moray.js
+++ b/test/lib/moray.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -79,26 +79,30 @@ function writeObjects(morayClient, bucketName, valueTemplate, nbObjects,
 
     var i;
 
-    var objectKeys = [];
+    var requests = [];
     for (i = 0; i < nbObjects; ++i) {
-        objectKeys.push(libuuid.create());
+        var newObjectValue = jsprim.deepCopy(valueTemplate);
+        var newObjectUuid = libuuid.create();
+        newObjectValue.uuid = newObjectUuid;
+        requests.push({
+            bucket: bucketName,
+            key: newObjectUuid,
+            operation: 'put',
+            value: newObjectValue,
+            option: {
+                etag: null
+            }
+        });
     }
 
-    vasync.forEachParallel({
-        func: function writeObject(objectUuid, done) {
-            var newObjectValue = jsprim.deepCopy(valueTemplate);
-            newObjectValue.uuid = objectUuid;
-            /*
-             * noBucketCache: true is needed so that when putting objects in
-             * moray after a bucket has been deleted and recreated, it doesn't
-             * use an old bucket schema and determine that it needs to update an
-             * _rver column that doesn't exist anymore.
-             */
-            morayClient.putObject(bucketName, objectUuid, newObjectValue, {
-                noBucketCache: true
-            }, done);
-        },
-        inputs: objectKeys
+    /*
+     * noBucketCache: true is needed so that when putting objects in
+     * moray after a bucket has been deleted and recreated, it doesn't
+     * use an old bucket schema and determine that it needs to update an
+     * _rver column that doesn't exist anymore.
+     */
+    morayClient.batch(requests, {
+        noBucketCache: true
     }, callback);
 }
 
-- 
2.21.0

